core/modules/parser/SqlSQL2.g

```diff
@@ -309,6 +309,11 @@ boost::shared_ptr<VoidTwoRefFunc> _qservFctSpecHandler;
 // ---------------
 // Top-level rules
 // ---------------
+// Qserv-supported statement, requires end of file, no statement chaining
+qserv_stmt : 
+      select_stmt (SEMICOLON)* EOF

```

Comment:
Yes, the semicolon is optional, and this is intentional. Jacek has noted to me that the semicolon is indeed optional, and is mainly useful as a statement separator